lib/copy/copy.js

```diff
@@ -104,21 +113,27 @@ function copyFile (srcStat, src, dest, opts, cb) {
 function copyFileFallback (srcStat, src, dest, opts, cb) {
   const rs = fs.createReadStream(src)
   rs.on('error', err => cb(err)).once('open', () => {
-    const ws = fs.createWriteStream(dest, { mode: srcStat.mode })
+    // explicitly not setting srcStat mode - will do that last
+    const ws = fs.createWriteStream(dest)
     ws.on('error', err => cb(err))
       .on('open', () => rs.pipe(ws))
-      .once('close', () => setDestModeAndTimestamps(srcStat, dest, opts, cb))
+      .once('close', () => setDestTimestampsAndMode(srcStat, src, dest, opts, cb))
   })
 }
 
-function setDestModeAndTimestamps (srcStat, dest, opts, cb) {
-  fs.chmod(dest, srcStat.mode, err => {
-    if (err) return cb(err)
-    if (opts.preserveTimestamps) {
-      return utimes(dest, srcStat.atime, srcStat.mtime, cb)
-    }
-    return cb()
-  })
+function setDestTimestampsAndMode (srcStat, src, dest, opts, cb) {
+  if (opts.preserveTimestamps) {
+    // The initial srcStat.atime cannot be trusted because it is modified by the read(2) system call
+    // (See https://nodejs.org/api/fs.html#fs_stat_time_values)
+    return fs.stat(src, (err, updatedSrcStat) => {
+      if (err) return cb(err)
+      return utimesMillis(dest, updatedSrcStat.atime, srcStat.mtime, (err2) => {
```

Comment:
same here as commented above.

Reply:
Fixed in [75756c4](https://github.com/jprichardson/node-fs-extra/pull/633/commits/75756c414d5542cd69789c41ab48dfc6268ae569)